entry.go

```diff
@@ -32,10 +69,117 @@ type Entry struct {
 	Level   Level
 	Time    time.Time
 	Message string
-	enc     Encoder
+	Caller  EntryCaller
+	Stack   string
+}
+
+// CheckWriteAction indicates what action to take after (*CheckedEntry).Write
+// is done.
+type CheckWriteAction int
+
+const (
+	// WriteThenNoop is the default behavior to do nothing speccial after write.
+	WriteThenNoop = CheckWriteAction(iota)
+	// WriteThenFatal causes a fatal os.Exit() after Write.
+	WriteThenFatal
+	// WriteThenPanic causes a panic() after Write.
+	WriteThenPanic
+)
+
+// CheckedEntry is an Entry together with an opaque Facility that has already
+// agreed to log it (Facility.Enabled(Entry) == true). It is returned by
+// Logger.Check to enable performance sensitive log sites to not allocate
+// fields when disabled.
+//
+// CheckedEntry references should be created by calling AddFacility() or
+// Should() on a nil *CheckedEntry. References are returned to a pool after
+// Write, and MUST NOT be retained after calling their Write() method.
+type CheckedEntry struct {
+	Entry
+	safeToWrite bool
+	should      CheckWriteAction
+	facs        []Facility
+	// TODO: we could provide static spac for the first N facilities to avoid
+	// allocations in common cases
+}
+
+// Write writes the entry to any Facility references stored, returning any
+// errors, and returns the CheckedEntry reference to a pool for immediate
+// re-use.
+func (ce *CheckedEntry) Write(fields ...Field) error {
```

Comment:
I want to point out to reviewers this subtle semantic change:
- prior `(*CheckedMessage).Write` was void return, it internally noted any error
- new `(*CheckedEntry).Write` returns the error

Is this okay or should I restore the prior behavior that users of `Logger.Check().Write()` don't have to handle errors?